### PR TITLE
changes for previous day latest

### PR DESCRIPTION
**Pull request by SahithiMy**
Signed-off-by: SahithiMy <sahithi.mylangam@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
Changed previous day latest to false for control index report 

### :chains: Related Resources
https://chefio.atlassian.net/jira/software/c/projects/STALWART/boards/361?modal=detail&selectedIssue=STALWART-166
### :+1: Definition of Done
Changed the day latest for control index structure
### :athletic_shoe: How to Build and Test the Change
rebuild components/compliance-service/
chef_load_compliance_scans
### :white_check_mark: Checklist

**All PRs** must tick these:

- [X] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [X] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [X] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [X] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [X] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [X] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [X] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [X] Tests added/updated? (all new code needs new tests)
- [X] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 1]
```diff
@@ -906,3 +906,31 @@ func (backend *ESClient) GetDocByReportUUId(ctx context.Context, data *relaxting
 	}
 	return &item, nil
 }
+
+func (backend *ESClient) SetDayLatestToFalseForControlIndex(ctx context.Context, controlId string, profileId string, mapping mappings.Mapping, index string, node_uuid string) error {
```
Comment (kalroy) [inline components/compliance-service/ingest/ingestic/ingestic.go:910]: We follow camelcase. Rename this parameter `node_uuid` to `nodeUuid`

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 2]
```diff
@@ -906,3 +906,31 @@ func (backend *ESClient) GetDocByReportUUId(ctx context.Context, data *relaxting
...
+	termQueryThisControl := elastic.NewTermsQuery("_id", GetDocIdByControlIdAndProfileID(controlId, profileId))
+
+	boolQueryDayLatest := elastic.NewBoolQuery().
+		Must(termQueryThisControl)
+
+	script := elastic.NewScript("def targets = ctx._source.nodes.findAll(node -> node.node_uuid == params.node_uuid); for(node in targets) { if(node.day_latest==true) {node.day_latest = false}} if (ctx._source.day_latest==true) {ctx._source.day_latest = false;}").Param("node_uuid", node_uuid)
```
Comment (kalroy) [inline components/compliance-service/ingest/ingestic/ingestic.go:916]: This line needs formatting. It is unreadable

**Review by kalroy — CHANGES_REQUESTED** [review]
The coverage is again a concern and we have a code smell too. Please keep a note of the file to be included in the unit test coverage.

**Review by dave-augustus — APPROVED** [review]

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 3]
```diff
@@ -990,3 +990,36 @@ func createScriptForAddingNode(node relaxting.Node) *elastic.Script {
...
+	}
+	if (ctx._source.day_latest==true) {
+		ctx._source.day_latest = false;
+	}`).Param("node_uuid", nodeId)
+	oneDayAgo := time.Now().Add(-24 * time.Hour)
+	time90daysAgo := time.Now().Add(-24 * time.Hour * 90)
+	// Getting all the indices
+	esIndexes, err := relaxting.IndexDates(relaxting.CompDailyControlIndexPrefix, time90daysAgo.Format(time.RFC3339), oneDayAgo.Format(time.RFC3339))
+	if err != nil {
+		logrus.Errorf("Cannot get indexes: %+v", err)
```
Comment (Dmaddu) [inline components/compliance-service/ingest/ingestic/ingestic.go:1015]: we need to do error handling properly, here we are just printing the error. In this case, we need to send the error to the caller.

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 4]
```diff
@@ -990,3 +990,36 @@ func createScriptForAddingNode(node relaxting.Node) *elastic.Script {
...
+	def targets = ctx._source.nodes.findAll(node -> node.node_uuid == params.node_uuid);
+	for(node in targets) { 
+		if(node.day_latest==true) {
+			node.day_latest = false
+		}
+	}
+	if (ctx._source.day_latest==true) {
+		ctx._source.day_latest = false;
+	}`).Param("node_uuid", nodeId)
+	oneDayAgo := time.Now().Add(-24 * time.Hour)
```
Comment (Dmaddu) [inline components/compliance-service/ingest/ingestic/ingestic.go:1010]: variable naming is not correct, please rename, if not simply pass the values to relaxting.IndexDates method and add the comments

File: components/compliance-service/ingest/ingestic/ingestic.go [hunk 5]
```diff
@@ -990,3 +990,36 @@ func createScriptForAddingNode(node relaxting.Node) *elastic.Script {
 	return elastic.NewScript("if (!(ctx._source.nodes instanceof Collection)) {ctx._source.nodes = [ctx._source.nodes];} ctx._source.nodes.add(params.node)").Params(params)
 
 }
+
+func (backend *ESClient) SetDayLatestToFalseForControlIndex(ctx context.Context, controlId string, profileId string, mapping mappings.Mapping, index string, nodeId string) error {
```
Comment (Dmaddu) [inline components/compliance-service/ingest/ingestic/ingestic.go:994]: please add a description for the method.

**Review by Dmaddu — CHANGES_REQUESTED** [review]

**Review by kalroy — APPROVED** [review]

**Pull request merged**